packages/dds/tree/src/index.ts

```diff
@@ -327,7 +327,7 @@ export {
 	test_RecursiveObject_base,
 	test_RecursiveObjectPojoMode,
 } from "./simple-tree/index.js";
-export { SharedTree, TreeFactory } from "./treeFactory.js";
```

Comment:
This change does not export the shared object class (I was the person driving the change to avoid doing that, which I already finished in the PR you linked). SharedTree is an instance of ISharedObjectKind (which is just our two factory APIs in a format compatible with container schema) as well as the interface type. So that is just exporting the factory and interface.

This PR is updating the code in this file use the public entry point to get the factory (SharedTree.getFactory) instead of the previously `@internal` (no longer exported) concreate factory class. Thus this change actually makes this file stop depending on any concreate classes from tree.